src/mpl/src/gpu/mpl_gpu_cuda.c

```diff
@@ -15,13 +15,29 @@ typedef struct gpu_free_hook {
     struct gpu_free_hook *next;
 } gpu_free_hook_s;
 
+static int gpu_initialized = 0;
+static int device_count = -1;
+static int max_dev_id = -1;
+
 static gpu_free_hook_s *free_hook_chain = NULL;
 
 static CUresult CUDAAPI(*sys_cuMemFree) (CUdeviceptr dptr);
 static cudaError_t CUDARTAPI(*sys_cudaFree) (void *dptr);
 
 static int gpu_mem_hook_init();
 
+int MPL_gpu_get_dev_count(int *dev_cnt, int *dev_id)
+{
+    int ret = MPL_SUCCESS;
+    if (!gpu_initialized) {
+        ret = MPL_gpu_init(&device_count, &max_dev_id);
+    }
+
+    *dev_cnt = device_count;
+    *dev_id = max_dev_id;
+    return MPL_SUCCESS;
```

Comment:
@KaimingOuyang you capture a return code in `ret` on line 33, but then always return MPL_SUCCESS here. Should we change to `return ret;`?